server/bundler/extensions-loader.js

```diff
@@ -0,0 +1,74 @@
+const fs = require( 'fs' );
+const path = require( 'path' );
+const camelCase = require( 'lodash' ).camelCase;
+const compact = require( 'lodash' ).compact;
+
+function getReducerPath( extensionDir ) {
+	return path.join( 'client', 'extensions', extensionDir, 'state', 'reducer.js' );
+}
+
+function generateReducerRequireString( extensionDir ) {
+	return `'${ camelCase( extensionDir ) }': require( 'extensions/${ extensionDir }/state/reducer' )\n`;
+}
+
+function generateExtensionsModuleString( reducerRequires ) {
+	return `module.exports = {
```

Comment:
I suppose it could. I'll think about this, especially in the context of the testing issue for this PR as well.